spec/requests/lead_providers/confirm_schools_spec.rb

```diff
@@ -46,4 +50,65 @@
       expect(flash[:success]).to be_present
     end
   end
+
+  describe "PUT /lead-providers/report-schools/confirm" do
+    let!(:cohort) { create :cohort, :current }
+
+    it "creates new partnerships" do
+      put "/lead-providers/report-schools/confirm"
+
+      expect(lead_provider.partnerships).to match_array(
+        schools.map do |school|
+          an_object_having_attributes(
+            class: Partnership,
+            cohort_id: cohort.id,
+            school_id: school.id,
+            lead_provider_id: lead_provider.id,
+            delivery_partner_id: delivery_partner.id,
+          )
+        end,
+      )
+    end
+
+    it "schedules partnership notifications" do
+      put "/lead-providers/report-schools/confirm"
+
+      schools.each do |school|
+        expect(PartnershipNotificationService.new).to delay_execution_of(:notify)
+          .with(an_object_having_attributes(
+                  class: Partnership,
+                  cohort_id: cohort.id,
+                  school_id: school.id,
+                  lead_provider_id: lead_provider.id,
+                  delivery_partner_id: delivery_partner.id,
+                ))
+      end
+    end
+
+    context "when a single partnership creation fails" do
+      let(:failing_school) { schools.sample }
+
+      def confirm!
+        put "/lead-providers/report-schools/confirm"
+      rescue StandardError # rubocop:disable Lint/SuppressedException
+      end
+
+      before do
+        allow(Partnership).to receive(:create!).and_call_original
+        allow(Partnership).to receive(:create!)
+          .with(hash_including(school_id: failing_school.id))
+          .and_raise StandardError.new(Faker::Lorem.sentence)
+      end
+
+      it "is not creating any Partnerships" do
+        expect { confirm! }.not_to change(Partnership, :count)
+      end
+
+      it "schedules no partnership notifications" do
+        confirm!
+
+        expect(PartnershipNotificationService.new).not_to delay_execution_of(:notify)
```

Comment:
Why do we need to be so specific? 

Can we do something like https://stackoverflow.com/a/27639157? Do we expect any other delayed jobs to be scheduled here?

Reply:
@kicferk1 So the problem with `enqueued_jobs` as described on SO is an ActiveJob creation. Basicaly how it works is ActiveJob is creting a new queue adapter per each test, simply recording all the enqueued jobs in an array. This means two things: 
1. `object.delay.method()` will not be recorded - this is pure DelayedJob syntax, going around ActiveJob
2. Even if we wrap execution in ActiveJob object, ActiveJob::TestQueueAdapter will not be aware of database transactions (which is really one of few advantages of delayed job over sidekiq). This test would give false negative

Reply:
Can we do something like:
```
expect(Delayed::Job.count).to equal(1)
expect(Delayed::Job.first.handler.job_class).to equal(:notify)
```
As I understand it this gets rid of ActiveJob problems because it goes straight for the actual implementation.